CONTRIBUTING.md

```diff
@@ -0,0 +1,167 @@
+
+```sh
+# Method 1: install the virtualenv package
+pip3 install --user virtualenv
+
+# Create a new virtual environment
+virtualenv -p python3 NAME_OF_YOUR_VENV
+# or even
+virtualenv -p $(which python3) NAME_OF_YOUR_VENV
+
+# Method 2: install the python3-venv package
+sudo apt install python3-venv # (Ubuntu)
+
+# Create a new virtual environment
+python3 -m venv NAME_OF_YOUR_VENV
+
+# Method 1 & 2: activate your new virtual env
+source NAME_OF_YOUR_VENV/bin/activate
+
+# Method 1 & 2: install Substra package inside your fresh new virtual environment
+pip3 install substra
+
+# Method 1 & 2: stop your virtual environment
+deactivate
+```
+
+Some of you might prefer to use [Anaconda](https://anaconda.org/), if so, you will be able to manage your virtual environment like this:
+
+```sh
+# Create a new virtual environment
+conda create --name NAME_OF_YOUR_VENV
+
+# Activate your new virtual environment
+conda activate NAME_OF_YOUR_VENV
+
+# Stop your virtual environment
+deactivate
+```
+
+Once inside your new virtual environment, you can install the project dependencies with the command `pip3 install -r requirements.txt`. What is installed inside a virtual environment is separated from your general Python setup.
+
+If you want to go futher, please refer to the [official documentation](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html).
+
+If you are looking for more Python Virtual Environment resources, you might be interested in this post from [Real Python](https://realpython.com/python-virtual-environments-a-primer/).
+
+Please note that you can then select your new virtual environment as any other Python interpreter in your favorite IDE and be able to use the new installed packages as if it were installed on your general Python setup.
+
+### 2.2 Python Enhancement Proposals (PEP)
+
+- [PEP8](https://pep8.org/)
+- [PEP Index](https://www.python.org/dev/peps/) & [Repository](https://github.com/python/peps)
+- Real Python: [How to Write Beautiful Python Code With PEP 8](https://realpython.com/python-pep8/)
+
+#### Black formatter
+
+This tool might seem a little bit radical, but it is based on PEPs and offers several possibilies of customization. It will help you learn and improve your code
+
+You don't have to take all the suggested modifications (with the help of `git diff`) but it is a good reference based on PEP rules that will ensure **validity**, **maintainability** and **readability** of the code: [Package Repository](https://github.com/psf/black)
+
+##### Installation
+
+```sh
+pip3 install black
+```
+
+##### Usage
+
+```sh
+# Inspect a file
+black FILE.py
+# Inspect files in the current folder
+black .
+```
+
+#### Flake8 linter
```

Comment:
What is the difference linter vs formatter ?

Reply:
I naively would say that a linter deals with the quality of the code while a formatter focuses on the structure/format of the doc, it doesn't "evaluate"